src/test/java/io/fabric8/maven/docker/util/DockerFileUtilTest.java

```diff
@@ -69,6 +69,17 @@ public void testMultiStageNamed() throws Exception {
         assertEquals(false, fromClauses.hasNext());
     }
 
+    @Test
+    public void testMultiStageWithArgs() throws Exception {
+        File toTest = copyToTempDir("Dockerfile_multi_stage_with_args");
+        Iterator<String> fromClauses = DockerFileUtil.extractBaseImages(
+                toTest, FixedStringSearchInterpolator.create()).iterator();
+
+        assertEquals("fabric8/s2i-java:latest", fromClauses.next());
+        assertEquals("busybox:latest", fromClauses.next());
+        assertEquals(false, fromClauses.hasNext());
+    }
+
```

Comment:
I would love to see more tests directly for the parsing methods with many corner cases. No need to create test Dockerfiles, but calling `extractArgs()` and `resolveImageTagFromArgs()` in a loop with different inputs (you could make the methods package private). But we need much more tests about different cases (like with ARG values, containing spaces, if this is allowed. Also check the Docker spec for the possible ARG variations).